core/chains/evm/client/chain_client_test.go

```diff
@@ -69,3 +69,250 @@ func TestChainClient_BatchCallContext(t *testing.T) {
+		err := ethClient.Dial(ctx)
+		require.NoError(t, err)
+
+		reqs := []client.TxSimulationRequest{
+			{
+				From: fromAddress1,
+				To:   &toAddress1,
+				Data: []byte("0x00"),
+			},
+			{
+				From: fromAddress2,
+				Data: []byte("0x01"),
+			},
+		}
+		err = ethClient.BatchCheckTxValidity(ctx, reqs)
+		require.NoError(t, err)
+		require.Empty(t, reqs[0].Error)
+		require.Empty(t, reqs[1].Error)
+	})
+
+	t.Run("returns zk out-of-counters error in request", func(t *testing.T) {
+		oocError := "not enough keccak counters to continue the execution"
+		mockRpc := newMockRpc(t)
+		mockRpc.On("BatchCallContext", mock.Anything, mock.IsType([]rpc.BatchElem{})).Run(func(args mock.Arguments) {
+			reqs := args.Get(1).([]rpc.BatchElem)
+			// Return proper result for a request
+			reqs[0].Result = `"0x100"`
+			// Return error for a request
+			reqs[1].Error = errors.New(oocError)
+		}).Return(nil).Once()
+
+		ethClient := client.NewChainClientWithMockedRpc(t, commonclient.NodeSelectionModeRoundRobin, time.Second*0, time.Second*0, testutils.FixtureChainID, mockRpc)
+		err := ethClient.Dial(ctx)
+		require.NoError(t, err)
+
+		reqs := []client.TxSimulationRequest{
+			{
+				From: fromAddress1,
+				To:   &toAddress1,
+				Data: []byte("0x00"),
+			},
+			{
+				From: fromAddress2,
+				To:   &toAddress1,
+				Data: []byte("0x01"),
+			},
+		}
+		err = ethClient.BatchCheckTxValidity(ctx, reqs)
+		require.NoError(t, err)
+
+		// No error for first request
+		require.Empty(t, reqs[0].Error)
+
+		// Out-of-counter error for second request
+		require.Equal(t, oocError, reqs[1].Error.Error())
+		require.Equal(t, true, reqs[1].Error.IsOutOfCounters())
+	})
+
+	t.Run("returns other errors in request", func(t *testing.T) {
+		errorMsg := "something went wrong"
+		mockRpc := newMockRpc(t)
+		mockRpc.On("BatchCallContext", mock.Anything, mock.IsType([]rpc.BatchElem{})).Run(func(args mock.Arguments) {
+			reqs := args.Get(1).([]rpc.BatchElem)
+			// Return proper result for a request
+			reqs[0].Result = `"0x100"`
+			// Return error for a request
+			reqs[1].Error = errors.New(errorMsg)
+		}).Return(nil).Once()
+
+		ethClient := client.NewChainClientWithMockedRpc(t, commonclient.NodeSelectionModeRoundRobin, time.Second*0, time.Second*0, testutils.FixtureChainID, mockRpc)
+		err := ethClient.Dial(ctx)
+		require.NoError(t, err)
+
+		reqs := []client.TxSimulationRequest{
+			{
+				From: fromAddress1,
+				To:   &toAddress1,
+				Data: []byte("0x00"),
+			},
+			{
+				From: fromAddress2,
+				To:   &toAddress1,
+				Data: []byte("0x01"),
+			},
+		}
+		err = ethClient.BatchCheckTxValidity(ctx, reqs)
+		require.NoError(t, err)
+
+		// No error for first request
+		require.Empty(t, reqs[0].Error)
+
+		// No Out-of-counter error for second request
+		require.Equal(t, errorMsg, reqs[1].Error.Error())
+	})
+
+	t.Run("returns the proper errors to the associated requests", func(t *testing.T) {
```

Comment:
I think this functionality is already tested in `TestChainClient_BatchCallContext`

Reply:
True. This is left over from when I had the request/batchElem matching which is removed now. I've deleted this in the latest commit.